guides/argument-semantics.md

```diff
@@ -0,0 +1,127 @@
+`job-iteration` overrides the `perform` method of `ActiveJob::Base` to allow for iteration. The `perform` method preserves all the standard calling conventions of the original, but the way the subsequent methods work might differ from what one expects from an ActiveJob subclass.
+
+The call sequence is usually 3 methods:
+
+`perform -> build_enumerator -> each_iteration|each_batch`
+
+In that sense `job-iteration` works like a framework (it calls your code) rather than like a library (that you call). When using jobs with parameters, the following rules of thumb are good to keep in mind.
+
+### Jobs without arguments
+
+Jobs without arguments do not pass anything into either `build_enumerator` or `each_iteration` except for the `cursor` which `job-iteration` persists by itself:
+
+```ruby
+class ArglessJob < ActiveJob::Base
+  include JobIteration::Iteration
+
+  def build_enumerator(cursor:)
+    # ...
+  end
+
+  def each_iteration(single_object_yielded_from_enumerator)
+    # ...
+  end
+end
+
+To spool the job:
```

Comment:
I'd change this and below to enqueue, [like the Active Job docs](https://guides.rubyonrails.org/active_job_basics.html#enqueue-the-job).